File: backend/DigitalLibrary.DAL/Entities/Order.cs
using DigitalLibrary.DAL.Enums;

namespace DigitalLibrary.DAL.Entities;

public class Order
{
    public int Id { get; set; }

    public int UserId { get; set; }

    public User User { get; set; } = null!;

    public int BookId { get; set; }

    public Book Book { get; set; } = null!;

    public string PhoneNumber { get; set; } = string.Empty;

    public DateTime? BorrowedAt { get; set; }

    public DateTime? DueDate { get; set; }

    public DateTime? ReturnedAt { get; set; }

    public OrderStatus Status { get; set; } = OrderStatus.Requested;

    public int? ManagerId { get; set; }

    public User? Manager { get; set; }

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
}

File: frontend/src/shared/api/order/types/order.ts
export type OrderStatus =
  | 'Requested'
  | 'Approved'
  | 'Borrowed'
  | 'Returned'
  | 'Rejected'
  | 'Overdue';

export interface Order {
  id: number;
  bookId: number;
  bookTitle: string;
  userId: number;
  userFullName: string;
  userEmail: string;
  phoneNumber: string;
  borrowedAt: string | null;
  dueDate: string | null;
  returnedAt: string | null;
  status: OrderStatus | string;
  managerId: number | null;
  managerFullName: string | null;
  createdAt: string;
}

export interface OrderRequestPayload {
  bookId: number;
  phoneNumber: string;
}
